core/services/relay/evm/functions/logpoller_wrapper.go

```diff
@@ -383,59 +326,321 @@ func (l *logPollerWrapper) getCurrentCoordinators(ctx context.Context) (common.A
+				CoordinatorContract: coordinator.address,
+			})
+		}
+		return requests, nil
+	}
+
+	if strings.Contains(coordinator.typeAndVersion, "Functions Coordinator v2") {
+		parsingContract, err := functions_coordinator.NewFunctionsCoordinator(coordinator.address, l.client)
+		if err != nil {
+			return nil, errors.Errorf("LogsToRequests: creating a contract instance for parsing failed")
+		}
+
+		for _, log := range requestLogs {
+			gethLog := log.ToGethLog()
+			oracleRequest, err := parsingContract.ParseOracleRequest(gethLog)
+			if err != nil {
+				l.lggr.Errorw("LogsToRequests: failed to parse a request log, skipping", "err", err)
+				continue
+			}
+
+			commitmentABI := abi.Arguments{
+				{Type: bytes32Type}, // RequestId
+				{Type: addressType}, // Coordinator
+				{Type: uint96Type},  // EstimatedTotalCostJuels
+				{Type: addressType}, // Client
+				{Type: uint64Type},  // SubscriptionId
+				{Type: uint32Type},  // CallbackGasLimit
+				{Type: uint72Type},  // AdminFee
+				{Type: uint72Type},  // DonFee
+				{Type: uint40Type},  // GasOverheadBeforeCallback
+				{Type: uint40Type},  // GasOverheadAfterCallback
+				{Type: uint32Type},  // TimeoutTimestamp
+				{Type: uint72Type},  // OperationFee
+			}
+			commitmentBytes, err := commitmentABI.Pack(
+				oracleRequest.Commitment.RequestId,
+				oracleRequest.Commitment.Coordinator,
+				oracleRequest.Commitment.EstimatedTotalCostJuels,
+				oracleRequest.Commitment.Client,
+				oracleRequest.Commitment.SubscriptionId,
+				oracleRequest.Commitment.CallbackGasLimit,
+				oracleRequest.Commitment.AdminFee,
+				oracleRequest.Commitment.DonFee,
+				oracleRequest.Commitment.GasOverheadBeforeCallback,
+				oracleRequest.Commitment.GasOverheadAfterCallback,
+				oracleRequest.Commitment.TimeoutTimestamp,
+				oracleRequest.Commitment.OperationFee,
+			)
+			if err != nil {
+				l.lggr.Errorw("LogsToRequests: failed to pack commitment bytes, skipping", err)
+			}
+
+			requests = append(requests, evmRelayTypes.OracleRequest{
+				RequestId:           oracleRequest.RequestId,
+				RequestingContract:  oracleRequest.RequestingContract,
+				RequestInitiator:    oracleRequest.RequestInitiator,
+				SubscriptionId:      oracleRequest.SubscriptionId,
+				SubscriptionOwner:   oracleRequest.SubscriptionOwner,
+				Data:                oracleRequest.Data,
+				DataVersion:         oracleRequest.DataVersion,
+				Flags:               oracleRequest.Flags,
+				CallbackGasLimit:    oracleRequest.CallbackGasLimit,
+				TxHash:              oracleRequest.Raw.TxHash,
+				OnchainMetadata:     commitmentBytes,
+				CoordinatorContract: coordinator.address,
+			})
+		}
+		return requests, nil
+	}
+
+	return nil, errors.Errorf("LogsToRequests: Unsupported Coordinator version %s", coordinator.typeAndVersion)
+}
+
+func (l *logPollerWrapper) logsToResponses(coordinator coordinator, responseLogs []logpoller.Log) ([]evmRelayTypes.OracleResponse, error) {
+	var responses []evmRelayTypes.OracleResponse
+
```

Comment:
Why not set the parsingContract here based on the version, then remove the 2 duplicated code blocks below?

Reply:
This is because we don't have a robust way to handle the two different types of wrappers here. Go is very strict about typing. We need a wrapper of wrappers of sorts. Something we will add to our tech debt and tackle soon.